api/v1beta1/clusterclass_types.go

```diff
@@ -391,6 +391,7 @@ type PatchSelectorMatch struct {
 
 	// MachineDeploymentClass selects templates referenced in specific MachineDeploymentClasses in
 	// .spec.workers.machineDeployments.
+	// Note: If set to an empty object it will match all MachineDeploymentClasses.
```

Comment:
How about also add a comment on `.PatchSelectorMatchMachineDeploymentClass.Names` that reads: `Note: If the list is empty then it sill match all MachineDeploymentClass`? 
Because, I think 
```
        matchResources:
          controlPlane: true
          machineDeploymentClass:
                name: []
```
will also pass the implemented logic and it will match all MachineDeploymentClasses. 

This is not blocking for the PR.

Reply:
You're right. Apart from the note, I have also added a unit test just to be on the safe side.